src/main/cpp/tests/cast_string.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cast_string.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/table_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <rmm/device_uvector.hpp>
+
+using namespace cudf;
+
+template <typename T>
+struct StringToIntegerTests : public test::BaseFixture {
+};
+
+TYPED_TEST_SUITE(StringToIntegerTests, cudf::test::IntegralTypesNotBool);
+
+TYPED_TEST(StringToIntegerTests, Simple)
+{
+  auto const strings = test::strings_column_wrapper{"1", "0", "42"};
+  strings_column_view scv{strings};
+
+  auto const result = spark_rapids_jni::string_to_integer(
+    data_type{type_to_id<TypeParam>()}, scv, false, rmm::cuda_stream_default);
+
+  test::fixed_width_column_wrapper<TypeParam> expected({1, 0, 42}, {1, 1, 1});
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected);
+}
+
+TYPED_TEST(StringToIntegerTests, Ansi)
+{
+  auto const strings =
+    test::strings_column_wrapper({"+1",
+                                  "-0",
+                                  "4.2",
+                                  "asdf",
+                                  "98fe",
+                                  "  00012",
+                                  ".--e-37602.n",
```

Comment:
There should be a test for a case where the only thing wrong is an extra decimal (e.g.: `3..5` or `3.54.0`.  There should also be tests for values with only trailing whitespace and both leading and trailing whitespace.